### PR TITLE
Update Hashicorp secrets backend setup + Enterprise secrets backend docs

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/347
Resolves https://github.com/astronomer/docs/issues/345

2-in-1 PR for Secrets Backend-related changes. The only new content is the updated Hashicorp Vault instruction. The only other changes are porting the Cloud content over to Enterprise and reformatting it accordingly.

### CODE REVIEW

File: cloud/secrets-backend.md [hunk 1]
```diff
@@ -47,10 +47,11 @@ Setting Airflow connections via secrets requires knowledge of how to generate Ai
...
-2. Configure your Astronomer project to pull the secret from Vault.
-3. Test the backend in a local environment.
-4. Deploy your changes to Astronomer Cloud.
+1. Create an AppRole which grants Astronomer minimal required permissions.
+2. Write a test Airflow variable or connection as a secret to your Vault server.
+3. Configure your Astronomer project to pull the secret from Vault.
+4. Test the backend in a local environment.
+5. Deploy your changes to Astronomer Cloud.
 
 ### Prerequisites
```
Comment (jwitz) [inline cloud/secrets-backend.md:56]: @fhoda Can you specifically review this section to make sure that I captured all of the details you described here? https://github.com/astronomer/docs/issues/345#issuecomment-1029184246

File: cloud/secrets-backend.md [hunk 2]
```diff
@@ -47,10 +47,11 @@ Setting Airflow connections via secrets requires knowledge of how to generate Ai
 
 This topic provides steps for how to use [Hashicorp Vault](https://www.vaultproject.io/) as a secrets backend for both local development and on Astronomer Cloud. To do this, you will:
 
-1. Write a test Airflow variable or connection as a secret to your Vault server.
-2. Configure your Astronomer project to pull the secret from Vault.
-3. Test the backend in a local environment.
-4. Deploy your changes to Astronomer Cloud.
+1. Create an AppRole which grants Astronomer minimal required permissions.
```
Comment (paolaperaza) [inline cloud/secrets-backend.md:50]:
```suggestion
1. Create an AppRole in Vault which grants Astronomer minimal required permissions.
```

File: cloud/secrets-backend.md [hunk 3]
```diff
@@ -61,15 +62,40 @@ To use this feature, you need:
...
-- Your Vault server's [Root Token](https://www.vaultproject.io/docs/concepts/tokens#root-tokens).
 - Your Vault Server's URL. If you're using a local server, this should be `http://127.0.0.1:8200/`.
 
 If you do not already have a Vault server deployed but would like to test this feature, we recommend that you either:
 
 - Sign up for a Vault trial on [Hashicorp Cloud Platform (HCP)](https://cloud.hashicorp.com/products/vault) or
 - Deploy a local Vault server via the instructions in [our Airflow Guide](https://www.astronomer.io/guides/airflow-and-hashicorp-vault).
 
-### Step 1: Write an Airflow Variable or Connection to Vault
+### Step 1: Configure Vault
```
Comment (paolaperaza) [inline cloud/secrets-backend.md:72]:
```suggestion
### Step 1: Create a Policy and AppRole in Vault
```

**Review by paolaperaza — APPROVED** [review]
LGTM @jwitz - I made a few minor suggestions but feel free to modify/merge. Make sure to copy those changes to Enterprise docs if you make em.

File: cloud/secrets-backend.md [hunk 4]
```diff
@@ -61,15 +62,40 @@ To use this feature, you need:
...
+
+To use Vault as a secrets backend, we recommend configuring a Vault role with a policy that grants only the minimum necessary permissions for Astronomer. To do this:
+
+1. [Create a Vault policy](https://www.vaultproject.io/docs/concepts/policies) with the following permissions:
+
+    ```hcl
+    path "secret/variables/*" {
```
Comment (fhoda) [inline cloud/secrets-backend.md:79]:
```suggestion
    path "secret/data/variables/*" {
```

File: cloud/secrets-backend.md [hunk 5]
```diff
@@ -61,15 +62,40 @@ To use this feature, you need:
...
+      capabilities = ["read", "list"]
+    }
+
+    path "secret/connections/*" {
```
Comment (fhoda) [inline cloud/secrets-backend.md:83]:
```suggestion
    path "secret/data/connections/*" {
```

File: cloud/secrets-backend.md [hunk 6]
```diff
@@ -105,18 +131,16 @@ apache-airflow-providers-hashicorp
 Then, add the following environment variables to your `Dockerfile`:
 
 ```text
-# Make sure to replace `<your-root-token>` and `<your-vault-url>` with your own values.
-ENV VAULT__ROOT_TOKEN="<your-root-token>"
-ENV VAULT__URL="<your-vault-url>"
+# Make sure to replace `<your-approle-id>` and `<your-approle-secret>` with your own values.
 ENV AIRFLOW__SECRETS__BACKEND="airflow.contrib.secrets.hashicorp_vault.VaultBackend"
-ENV AIRFLOW__SECRETS__BACKEND_KWARGS='{"url":$VAULT__URL,"token": $VAULT__ROOT_TOKEN,"connections_path": "connections","variables_path": "variables","kv_engine_version":1}'
+ENV AIRFLOW__SECRETS__BACKEND_KWARGS='{"connections_path": "airflow/connections", "variables_path": null, "config_path": null, "url": "https://vault.vault.svc.cluster.local:8200", "auth_type": "approle", "role_id":"<your-approle-id>", "secret_id":"<your-approle-secret>"}'
```
Comment (fhoda) [inline cloud/secrets-backend.md:136]:
```suggestion
ENV AIRFLOW__SECRETS__BACKEND_KWARGS='{"connections_path": "airflow/connections", "variables_path": null, "config_path": null, "url": $VAULT__URL, "auth_type": "approle", "role_id":"<your-approle-id>", "secret_id":"<your-approle-secret>"}'
```

File: cloud/secrets-backend.md (hunk 1 above)
Comment (fhoda) [reply cloud/secrets-backend.md:56]: @jwitz please see suggestions/comments

File: cloud/secrets-backend.md (hunk 6 above)
Comment (fhoda) [reply cloud/secrets-backend.md:136]: The values `role_id` and `secret_id` should probably be stored and referenced as environment variables. Very much like the original guide that used a root token. 

so would look something like
```
role_id": $VAULT_ROLE_ID, "secret_id": $VAULT_SECRET_ID
```

**Comment by jwitz** [comment]
Thank you for the review @fhoda ! Ended up keeping the same Dockerfile-based config for local testing in order to simplify steps, and because this config will have to be updated anyways once the user deploys to Astronomer. If there's a better env var build that you can see being used in production then I'm happy to edit both the local / Astronomer setup steps with that build

**Pull request merged**